core/src/main/java/org/mule/processor/chain/SubflowInterceptingChainLifecycleWrapper.java

```diff
@@ -52,6 +56,28 @@ public void addMessageProcessorPathElements(MessageProcessorPathElement pathElem
     }
 
     @Override
+    public MuleEvent process(MuleEvent event) throws MuleException
+    {
+        Collection<MessageProcessingFlowStackManager> flowStackManagers = event.getMuleContext().getRegistry().lookupObjects(MessageProcessingFlowStackManager.class);
```

Comment:
Are you expecting to have multiple MessageProcessingFlowStackManager instances? Why not to have this on the muleContext?